src/test/compile-fail/block-must-not-have-result-do.rs

```diff
@@ -10,6 +10,14 @@
 
 fn main() {
     loop {
-        true //~  ERROR mismatched types
+        true
+        //~^ ERROR mismatched types
+        //~| ERROR mismatched types
+        //~| NOTE: possibly missing `;` here?
+        //~| NOTE: expected (), found bool
+        //~| NOTE: expected type `()`
+        //~| NOTE: expected type `()`
```

Comment:
I didn't dig too deeply into this [when I noticed it](https://github.com/rust-lang/rust/pull/36409#issuecomment-250346774), but we should probably create a ticket for it. Could the same DiagnosticBuilder be emitter multiple times? Maybe that's what's happening...